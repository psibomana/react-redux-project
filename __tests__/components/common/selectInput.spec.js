import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SelectInput from '../../../src/components/common/selectInput';

describe('The Select Input Common Component', () => {
  it('should render as expected', () => {
    const options = [
      {id: 1, firstName: "Option First Name 1", lastName: "Option Last Name 1"},
      {id: 2, firstName: "Option First Name 2", lastName: "Option Last Name 2"},
      {id: 3, firstName: "Option First Name 3", lastName: "Option Last Name 3"}
    ]
    const wrapper = shallow(
      <SelectInput
        id="Item ID"
        name="Item Name"
        label="Item Label"
        value="Item Value"
        defaultOption="Select Default Option"
        options={options}
        onChange={() => {console.log("Select Input value changed")}}
        error="Error"/>);

    expect(toJson(wrapper)).toMatchSnapshot();
  })
});
