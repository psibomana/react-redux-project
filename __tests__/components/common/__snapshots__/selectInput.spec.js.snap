// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Select Input Common Component should render as expected 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="Item Name"
  >
    Item Label
  </label>
  <div
    className="field"
  >
    <select
      className="form-control"
      name="Item Name"
      onChange={[Function]}
      value="Item Value"
    >
      <option
        value=""
      >
        Select Default Option
      </option>
      <option
        key="1"
        value="Option First Name 1 Option Last Name 1"
      >
        Option First Name 1 Option Last Name 1
      </option>
      <option
        key="2"
        value="Option First Name 2 Option Last Name 2"
      >
        Option First Name 2 Option Last Name 2
      </option>
      <option
        key="3"
        value="Option First Name 3 Option Last Name 3"
      >
        Option First Name 3 Option Last Name 3
      </option>
    </select>
    <div
      className="alert alert-danger"
    >
      Error
    </div>
  </div>
</div>
`;
